cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -44,18 +44,19 @@ __launch_bounds__(block_size) __global__
                            mutable_column_device_view out,
                            size_type* __restrict__ const valid_count)
 {
-  size_type const tid            = threadIdx.x + blockIdx.x * block_size;
-  int const warp_id              = tid / warp_size;
+  auto tidx                      = cudf::detail::grid_1d::global_thread_id();
+  auto const stride              = cudf::detail::grid_1d::grid_stride();
+  int const warp_id              = tidx / warp_size;
   size_type const warps_per_grid = gridDim.x * block_size / warp_size;
 
   // begin/end indices for the column data
-  size_type begin = 0;
-  size_type end   = out.size();
+  size_type const begin = 0;
+  size_type const end   = out.size();
   // warp indices.  since 1 warp == 32 threads == sizeof(bitmask_type) * 8,
   // each warp will process one (32 bit) of the validity mask via
   // __ballot_sync()
-  size_type warp_begin = cudf::word_index(begin);
-  size_type warp_end   = cudf::word_index(end - 1);
+  size_type const warp_begin = cudf::word_index(begin);
+  size_type const warp_end   = cudf::word_index(end - 1);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Should any of these use `thread_index_type` instead of `size_type`?

Reply:
This is a question that I had as well.  I was wondering if we should convert all these to `thread_index_type`.

I suppose `warp_curr` is less likely to overflow at line#87, so maybe it's fine as a `size_type`?

Reply:
If you trace the types, the value of `warp_cur` in `while (warp_cur <= warp_end) {` is the one to watch. Theoretically this value could be `thread_index_type`. However, it's compared to `warp_end` which is `size_type` (its maximum value is `std::numeric_limits<cudf::size_type>::max() / cudf::detail::warp_size`). So really I don't think we need to do anything here, otherwise we'll be casting up some computations "for safety reasons" only to unsafely downcast in a comparison.